src/common/functions/src/scalars/expressions/cast_from_string.rs

```diff
@@ -103,9 +106,26 @@ pub fn string_to_timestamp(date_str: impl AsRef<[u8]>, tz: &Tz) -> Option<DateTi
     let s = std::str::from_utf8(date_str.as_ref()).ok();
     if let Some(s) = s {
```

Comment:
These codes are duplicated by `read_timestamp_text`.

I think we just need to keep `read_timestamp_text`  are call this function in `cast_from_string`

Reply:
> These codes are duplicated by `read_timestamp_text`.
> 
> I think we just need to keep `read_timestamp_text` are call this function in `cast_from_string`

Yes, these codes are duplicated. And I will fix the bug in read_timestamp_text and next to resolve this.